src/main/groovy/edu/wpi/first/wpilib/versioning/WPILibVersioningPlugin.groovy

```diff
@@ -66,8 +69,41 @@ class WPILibVersioningPlugin implements Plugin<Project> {
             println "No version number generated."
             return ''
         }
+
         Grgit git = Grgit.open(currentDir: (Object)gitDir.absolutePath)
+        // Get the tag given by describe
         String tag = git.describe()
+
+        // Get a list of all tags
+        List<Tag> tags = git.tag.list()
+
+        Tag describeTag = null
+
+        // Find tag hash that starts with describe
```

Comment:
Why is this whole rigmarole necessary?

Reply:
Which part?  The describe part?  Or the sort by date part?

Reply:
grgit's describe returns the oldest tag attached to a commit if there are multiple tags pointing to the same commit. Not the newest tag.

Reply:
Ah, got it. Thanks. Consider an update to put this description in a comment.

Reply:
Yeah I'll do that. It just has been causing maven to overwrite builds, so it needed to be fixed.